### PR TITLE
extract defaults to member initialization, do some more value checking, ...

### PR DESCRIPTION
**Pull request by markushausammann**
...other details

### CODE REVIEW

File: src/DocRaptor.php [hunk 1]
```diff
@@ -71,10 +67,19 @@ public function setDocumentUrl($document_url)
...
+        $allowedValues = array('xls', 'xlsx', 'pdf');
+
+        if (! in_array(strtolower(trim($document_type)), $allowedValues)) {
+            throw new Exception('Value not accepted by method.');
+        }
+
         $document_type = strtolower($document_type);
+
+        //@todo why is xls cast to pdf?
         $this->type = $document_type == 'pdf' || $document_type == 'xls' ? $document_type : 'pdf';
```
Comment (janxious) [inline src/DocRaptor.php]: This was a null check, basically. You can remove this now given you are doing the check above?

File: src/DocRaptor.php [hunk 2]
```diff
@@ -71,10 +67,19 @@ public function setDocumentUrl($document_url)
     /**
      * @param string $document_type
      * @return $this
+     * @throws Exception
      */
     public function setDocumentType($document_type)
     {
+        $allowedValues = array('xls', 'xlsx', 'pdf');
```
Comment (janxious) [inline src/DocRaptor.php:74]: Constant?

File: src/DocRaptor.php (hunk 1 above)
Comment (markushausammann) [inline src/DocRaptor.php]: Ah, I read it wrongly anyways, yes, it can be removed.

File: src/DocRaptor.php [hunk 3]
```diff
@@ -155,11 +160,12 @@ public function setBaseUrl($base_url)
      *
      * @param bool|string $filename
      * @return bool|mixed
+     * @throws Exception
      */
     public function fetchDocument($filename = false)
     {
         if ($this->api_key != '') {
-            $url = $this->url_protocol . '://docraptor.com/docs?user_credentials=' . $this->api_key;
+            $url = $this->url_protocol . '://'. $this->api_url . '?user_credentials=' . $this->api_key;
```
Comment (janxious) [inline src/DocRaptor.php:165]: Is `.` preferred over string interpolation?

Comment (markushausammann) [inline src/DocRaptor.php:165]: No, actually this should be written with `sprintf()`. I'll do that.

File: src/DocRaptor.php [hunk 4]
```diff
@@ -185,12 +191,14 @@ public function fetchDocument($filename = false)
...
             curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
+            $result = curl_exec($ch);
 
-            if ($result = curl_exec($ch)) {
+            if ($result) {
                 $http_code = curl_getinfo($ch, CURLINFO_HTTP_CODE);
 
                 if ($http_code != 200) {
-                    return False;
+                    //@todo silently erroring out is probably not the best way here
```
Comment (janxious) [inline src/DocRaptor.php:197]: I agree. Maybe print the HTTP code and any text DR sends back?

File: src/DocRaptor.php (hunk 3 above)
Comment (janxious) [inline src/DocRaptor.php:165]: Another possibility: api key is in the `$fields` array below and gets handled by `http_build_query`

File: src/DocRaptor.php (hunk 2 above)
Comment (markushausammann) [inline src/DocRaptor.php:74]: Array constants exist only from PHP 5.6 but we can define single constants for the formats. Thoughts?

Comment (janxious) [inline src/DocRaptor.php:74]: I don't have any idea what adoption is for versions. This is fine as-is if 5.6 is a non-starter for a lot of people.

Comment (markushausammann) [inline src/DocRaptor.php:74]: Adoption is pretty slow but many will skip right to PHP 7 when adoption should speed up (I hope). Let's leave it for the moment.

File: src/DocRaptor.php (hunk 4 above)
Comment (markushausammann) [inline src/DocRaptor.php:197]: As in other cases I would throw an exception with the status code and message from DR then consumer can deal with it. Printing just forces them to extend or core hack.

**Comment by markushausammann** [comment]
Can we merge the current state? I'll change the two open points in another PR.

**Pull request merged**